lib/src/list_mutations.dart

```diff
@@ -172,14 +176,39 @@ SourceEdit _insertInBlockList(
 
   if (index == list.length) return _appendToBlockList(yamlEdit, list, item);
 
-  final formattedValue = _formatNewBlock(yamlEdit, list, item);
+  var (formattedValue, indent) = _formatNewBlock(yamlEdit, list, item);
 
   final currNode = list.nodes[index];
   final currNodeStart = currNode.span.start.offset;
   final yaml = yamlEdit.toString();
-  final start = yaml.lastIndexOf('\n', currNodeStart) + 1;
 
-  return SourceEdit(start, 0, formattedValue);
+  var currSequenceOffset = yaml.lastIndexOf('-', currNodeStart - 1);
+
+  final (isNested, offset) = _isNestedInBlockList(currSequenceOffset - 1, yaml);
+
+  /// We have to get rid of the left indentation applied by default
+  if (isNested && index == 0) {
+    // Give the previous first element its indent
+    formattedValue = formattedValue.trimLeft() + indent;
+  }
+
+  return SourceEdit(offset, 0, formattedValue);
+}
+
+/// Checks if the [YamlNode]'s list is directly nested within another list
+(bool isNested, int offset) _isNestedInBlockList(int start, String yaml) {
+  if (start < 0) return (false, 0);
+
+  final newLineStart = yaml.lastIndexOf('\n', start);
+  final seqStart = yaml.lastIndexOf('-', start);
+
+  /// Anytime our '-' is non-existent, use '\n' as source of truth. Also, if
+  /// the new line is closer
```

Comment:
What does this mean, when does this happen? Can it happen?

```suggestion
  /// If '\n' comes before '-' then we're not inside a nested block list.
```

Reply:
Yeah, it does since we are scanning backwards. I think you covered it in the last comment. Basically if:
1. Index of `-` is `-1` or less than `\n` index (even if `\n` index is `-1`), we always assume we inserting it i.e. `+1`.
2. Otherwise, we insert after the last `-` i.e. `+2 inclusive of space after if nested in list.

Reply:
nit: Should we update the comment or not?

Reply:
I'll make it clearer.